### ISSUE TITLE
Module can be uninstalled even if you don't click to uninstall them

### ISSUE THREAD

**Issue by bealdav** [odoo/odoo#14452]
### Impacted versions: V10 (tested on runbot), and V8 at least (I think all versions)
A closed tab in your browser can bring you to uninstall any module

### Steps to reproduce:
- goto runbot v10
- goto module lunch (or any module) in form view
- duplicate previous form view in an other tab
- click on Uninstall button: uninstall wizard confirmation appears
- close the tab
- goto cloned tab
- refresh tab: lunch module state is 'to be removed' instead of 'installed'

### Current behavior:
- lunch module state is now 'to be removed' instead of 'installed'.
- a future update of the erp (on any other module) will move lunch state to 'uninstalled', then you loose all your sandwich with this behavior

### Expected behavior:
- Odoo should only uninstall modules after the confirmation of the user
- Module state shouldn't change before uninstall wizard confirmation appears: only after confirmation of the wizard

### Credits
Sylvain LEGAL discovered this bug

**Comment by Yenthe666** [comment]
@mart-e and @mart-e it is not a normal case but it has a MAJOR impact if you should ever run into this. Can we look for a fix as soon as possible?

**Comment by mart-e** [comment]
Hi @Yenthe666 and @Yenthe666 ! ;-)

Thanks @bealdav for the report. You are right, so far, when pressing the uninstall button, the modules are put in "to uninstall" state and only the "cancel" button stops this process. This is why it is written in the wizard:

> If you wish to cancel the process, press the cancel button below

However, I know, people don't read warning messages, and some will close the window (even though this is a rather odd). So it could definitely be improved.

I guess we could move the downstream dependencies computation to the wizard and avoid going by to this 2 steps which I think is just to pass information and compute it only once.

**Comment by bealdav** [comment]
@mart-e  Any PR planned about it ?

**Comment by JKE-be** [comment]
Hello @bealdav 

It is only a question of usability and not really a bug... So it is not the priority today !
As explain by @mart-e, the warning is clear... the code do what the warning explains... the problem is to be sure that the user read the warning... and maybe confirm the warning to be able to dismiss the warning !
But yes, we have a task to improve it, but not the priority in our pipe.

Best regards

JKE

**Comment by bealdav** [comment]
@JKE-be  Thanks for your answer.
I understand you have priorities. 
But here we have a real risk to loose data with only Escape key by example. User can't expect to this issue with expect. It's not only usability
And this issue can have a really high cost for the enterprise.
I hope Odoo SA have a good insurance to protect it from paid customer claims or procedures.
I wish you there'll no problem with that, I hope sincerely.

**Comment by JKE-be** [comment]
>  I hope Odoo SA have a good insurance to protect it from paid customer claims or procedures.

When an user click on uninstall, I don't see why Odoo SA should be responsible of the data deleted by the uninstallation of the module.

But I'm agree, that the user misunderstanding can be done too easily, and the warning could be more explicit. It is just a question of time...

**Comment by bealdav** [comment]
> When an user click on uninstall, I don't see why Odoo SA should be responsible of the data deleted by the uninstallation of the module.

If you don't confirm uninstall popup, and uninstall happens, I think some advocates could see and maybe convince influent people.

Note: I don't want claim anything myself if my customer loose data (it happened before and I assumed myself), but it's just to help Odoo to prevent problems in future.

**Comment by JKE-be** [comment]
When you click to uninstall, it mark module and depends as 'to uninstall'.

We show a popup after to explains the impact of your previous action 'mark a to uninstall' that has been already done !

When you click, there are no confirm dialog ! 

But we offer to you the ability to undo the 'to uninstall' if you click on "Cancel'.

> This operation will permanently erase all data currently stored by the modules!
> **If you wish to cancel the process, press the cancel button below**

So if you don't press 'Cancel', you don't 'cancel the process'
In any case we are agree that it is not 100% clear, but It is not a bug, and you can uninstall your module and call your advocates and we will see ;) 

Sorry, but we spend more time to speaks about it on github that the time to provide a PR...
It just need to be improved and maybe have a 'confirm dialog' and not a 'cancel dialog'.

Best regards

JKE

**Comment by Yenthe666** [comment]
@JKE-be 

> When an user click on uninstall, I don't see why Odoo SA should be responsible of the data deleted by the uninstallation of the module.

That is a thin line to walk on. While the user does click on the `uninstall` button (perhaps by mistake, a misclick is easly done) it shouldn't drop the module unless you click the confirm button. What if we turn this around? As long as I haven't clicked on `confirm`, did I really want to drop this module?
I agree with you that this is not a very big thing **if** there is no data loss. However, the fact that all your data is dropped when you've made this mistake makes this a problem / bug in my opinion.
Data loss is something that shouldn't ever happen, I think we can all agree on that.

Either way just wanted to drop my five cents, just keep us updated if a fix is provided. I know you guys are busy.

**Comment by mart-e** [comment]
Hello guys,

As we said above, it can be improved but it is still in the "nice to have" category. If somebody submit a PR, I would gladly review it.

**Comment by bealdav** [comment]
@JKE-be you're right, it's not a real bug, but it can lead to very uncomfortable situation for everybody.

Then I think you should reconsider the priority on this base. 
I also know you are very busy to do some really good job, version after versions, but data loss here....

**Comment by pedrobaeza** [comment]
I think this is attended through the confirmation box, isn't it?

**Comment by JKE-be** [comment]
it has been improved meanwhile.

Now, it is a confirmation dialog...

Thanks for the reporting

**Comment by JKE-be** [comment]
![img_1975](https://user-images.githubusercontent.com/4848797/30004452-b740b8a6-90cf-11e7-86d6-d820cc7e4896.JPG)
